### PR TITLE
Pin 3022 [BE] Nel listing delle finalità lato erogatore devono essere escluse quelle in stato DRAFT

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
Added new parameter to excludeDraft from list purposes

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala [hunk 1]
```diff
@@ -119,7 +119,11 @@ object ReadModelQueries {
     // Note: the filter works on the assumption that if a version in Draft exists, it is the only version in the Purpose
     val versionsFilter = Filters.or(
       Filters
-        .and(Filters.eq("data.consumerId", requesterId.toString), Filters.eq("data.versions.state", Draft.toString)),
+        .and(
+          Filters.eq("data.consumerId", requesterId.toString),
+          Filters.ne("eservice.data.producerId", requesterId.toString),
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: what happens if I'm both the producer and the consumer, and I'm doing this request as the consumer to see my draft purposes?

Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: I asked to Carmine some consumer ids to test the different behaviour with or without this predicate.

consumerId: 5ec5dd81-ff71-4af8-974b-4190eb8347bf (consumer is also producer) without this predicate 7 results, with this predicate 5 results seems that exclude services for which it also is the producer.

consumerId: 05c7ad1f-84ca-41f5-932a-9cb1e0187e75 (consumer is not producer ) with or without the predicate always 7 results.

It seems to work!

Comment (galales) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: When the user is on the producer view is perfect, but what if the first consumer (`5ec5dd81-ff71-4af8-974b-4190eb8347bf`) access the list of its own purposes, which contains a Draft version for an EService that it owns?
Isn't the Draft purpose hidden in this case?

Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: We have three options:

1) User does not see his purposes
2) We need an additional parameter from BFF
3) We need to split endpoint

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: IMHO 1 it's a no-go. Both 2 and 3 will affect FE devs user experience, maybe it's worth asking them?

Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: Yes, Alessio decided to change splitting the BFF endpoint (the former for producer and the latter for the consumer) and passing a further parameter _excludeDraft_ to achieve the aim of excluding the DRAFT purposes from the query

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: Ok, that makes sense. I'll review the corresponding BFF Pr then. Thanks.

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -80,6 +80,10 @@ paths:
               $ref: '#/components/schemas/PurposeVersionState'
             default: []
           explode: false
+        - in: query
+          name: excludeDraft
+          schema:
+            type: boolean
```
Comment (galales) [inline src/main/resources/interface-specification.yml:86]: You could add a `default: false` to remove the optionality in the code.
This way it is also clear at spec level which is the default behaviour

File: src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala [hunk 3]
```diff
@@ -114,14 +118,16 @@ object ReadModelQueries {
...
+  private def listPurposesAuthorizationFilters(excludeDraft: Option[Boolean]): Bson = {
+    // Exclude draft purposes only if requested
     // Note: the filter works on the assumption that if a version in Draft exists, it is the only version in the Purpose
-    val versionsFilter = Filters.or(
-      Filters
-        .and(Filters.eq("data.consumerId", requesterId.toString), Filters.eq("data.versions.state", Draft.toString)),
-      Filters.ne("data.versions.state", Draft.toString)
-    )
+    val versionsFilterWithoutDraft: Bson = Filters.and(Filters.ne("data.versions.state", Draft.toString))
+    val versionsFilterWithDraft: Bson    = Filters.and(Filters.eq("data.versions.state", Draft.toString))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala]: a request with `excludeDraft == false` should not force the presence of a Draft, but include all versions despite the state.
I would expect this filter to be
```suggestion
    val versionsFilterWithDraft: Bson    = Filters.empty
```

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/resources/interface-specification.yml (hunk 2 above)
Comment (TonioGela) [reply src/main/resources/interface-specification.yml:86]: +1

**Review by TonioGela — DISMISSED** [review]
Same comment as galales

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**